hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientClusterServiceImpl.java

```diff
@@ -146,7 +146,7 @@ public Client getLocalClient() {
         InetSocketAddress inetSocketAddress = connection != null ? connection.getLocalSocketAddress() : null;
         ClientPrincipal principal = cm.getPrincipal();
         final String uuid = principal != null ? principal.getUuid() : null;
-        return new ClientImpl(uuid, inetSocketAddress);
+        return new ClientImpl(uuid, inetSocketAddress, client.getName(), client.getClientConfig().getAttributes());
```

Comment:
Probably best to make a copy of the clientConfig.attributes and wrap it in an immutable map so:
1) you are protected against accidental modification. And you don't need to make ad hoc wrappings.
2) if a user would make mutation on the client config after he has created the client, it will not to funky behavior.